library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -0,0 +1,1208 @@
+        if (viewDim >= contentDim) {
+            // Content is smaller, we are showing some boundary.
+            // We can't move in any direction (but we can overScroll).
+            mTemp[0] = currentPan + fix
+            mTemp[1] = currentPan
+            mTemp[2] = currentPan + fix
+        } else {
+            // Content is bigger, we can move.
+            // in this case minPan + viewDim = contentDim
+            mTemp[0] = -(contentDim - viewDim)
+            mTemp[1] = currentPan
+            mTemp[2] = 0
+        }
+        return fix != 0
+    }
+
+    private fun startFling(@ScaledPan velocityX: Int, @ScaledPan velocityY: Int): Boolean {
+        if (!setState(FLINGING)) return false
+
+        // Using actual pan values for the scroller.
+        // Note: these won't make sense if zoom changes.
+        var overScrolled: Boolean = computeScrollerValues(true)
+        @ScaledPan val minX = mTemp[0]
+        @ScaledPan val startX = mTemp[1]
+        @ScaledPan val maxX = mTemp[2]
+        overScrolled = overScrolled or computeScrollerValues(false)
+        @ScaledPan val minY = mTemp[0]
+        @ScaledPan val startY = mTemp[1]
+        @ScaledPan val maxY = mTemp[2]
+
+        val go = overScrolled || mOverScrollHorizontal || mOverScrollVertical || minX < maxX || minY < maxY
+        if (!go) {
+            setState(NONE)
+            return false
+        }
+
+        @ScaledPan val overScrollX = if (mOverScrollHorizontal) currentOverScroll else 0
+        @ScaledPan val overScrollY = if (mOverScrollVertical) currentOverScroll else 0
+        LOG.i("startFling", "velocityX:", velocityX, "velocityY:", velocityY)
+        LOG.i("startFling", "flingX:", "min:", minX, "max:", maxX, "start:", startX, "overScroll:", overScrollY)
+        LOG.i("startFling", "flingY:", "min:", minY, "max:", maxY, "start:", startY, "overScroll:", overScrollX)
+        mFlingScroller.fling(startX, startY,
+                velocityX, velocityY,
+                minX, maxX, minY, maxY,
+                overScrollX, overScrollY)
+
+        container.post(object : Runnable {
+            override fun run() {
+                if (mFlingScroller.isFinished) {
+                    setState(NONE)
+                } else if (mFlingScroller.computeScrollOffset()) {
+                    @ScaledPan val newPanX = mFlingScroller.currX
+                    @ScaledPan val newPanY = mFlingScroller.currY
+                    // OverScroller will eventually go back to our bounds.
+                    applyScaledPan(newPanX - scaledPanX, newPanY - scaledPanY, true)
+                    container.postOnAnimation(this)
+                }
+            }
+        })
+        return true
+    }
+
+    /**
+     * Set a container to perform transformations on.
+     * Any previously set container will no longer be modified.
+     *
+     * @param newContainer view
+     */
+    fun setContainer(newContainer: View) {
```

Comment:
Already discussed in other comment. & let's not make this public, use internal

Reply:
done